### ISSUE TITLE
changing dap-mode locals cap

### ISSUE THREAD

**Issue by bridgesense** [emacs-lsp/dap-mode#667]
Sorry, for the newbie question.  I didn't see an IRC channel for LSP or any reference to this setting in the documentation.

I'm using dap-mode with PHP and Xdebug.  My xdebug.var_display_max_data option is set to "-1".  I'm still getting a variable cap at around 120 in the locals reference.

Where do we set the display caps for the references in the Locals window for dap-mode?  Is that cap coming from another emacs module?

**Comment by yyoncho** [comment]
We have discord here: https://discord.com/invite/swuxy5AAgT

Can you check if  dap-ui-variable-length works for you.

**Comment by bridgesense** [comment]
Thanks for reminding me about that.  I did discover that I had already set it at 10000 my configuration.  I can confirm that is what it currently is set at with describe-variable.  It doesn't appear to have any effect.

Running the latest Ubuntu install of Emacs 27.1

**Comment by yyoncho** [comment]
Can you do: `(setq dap-print-io t)` and then go to the messages buffer and check if the server message has  the data you want to show. If it doesn' then can you show me the server message and a screenshot of how it is displayed in the UI.

**Comment by bridgesense** [comment]
I wish I could, but the data is sensitive information. 

My message buffer is getting cropped, too.  I set the  eval-expression-print-length to nil, but I'm only getting a portion of the readout.

Is there something specific you are looking for in the message buffer that I can look for?

**Comment by yyoncho** [comment]
> My message buffer is getting cropped

You mean that the message does not have the whole data? 

I want to find out who is cutting the data, the UI or the server.

Also you can still create a function foo with one param "bar" with long value and send me the data for that fuction (it won't have antyhing sensitive)

**Comment by yyoncho** [comment]
There is `message-log-max`.

**Comment by bridgesense** [comment]
Here's a code snippet:

![Same Code Snippet](https://i.imgur.com/c3XQywt.png)

It shows all 251 items  in the locals reference:

![Locals reference correct variable size](https://i.imgur.com/inyEZxh.png)

Interesting though, it cuts off after 100 records:

![only 100 items showing](https://i.imgur.com/946k4k6.png)

Thanks, in advance for your help in probing this issue!

**Comment by yyoncho** [comment]
Ok, I found the issue. I will try to provide a fix tomorrow my time.

**Comment by yyoncho** [comment]
Now you can use something like `(setq dap-ui-default-fetch-count 100)`. FWIW the real solution is to create paging here.

**Comment by yyoncho** [comment]
JFYI don't upgrade treemacs when upgrading dap-mode because latest version of lsp-mode won't work with latest treemacs

**Comment by bridgesense** [comment]
> JFYI don't upgrade treemacs when upgrading dap-mode because latest version of lsp-mode won't work with latest treemacs

Ouch!  OK, I had already updated that a while back.  Which version should I roll back to?

Hate to see this broken for fresh emacs installs.

**Comment by bridgesense** [comment]
> JFYI don't upgrade treemacs when upgrading dap-mode because latest version of lsp-mode won't work with latest treemacs

Which version does it work with, so I know what to roll back to?

**Comment by yyoncho** [comment]
@bridgesense https://github.com/Alexander-Miller/treemacs/issues/677#issuecomment-1245116779

**Comment by bridgesense** [comment]
Sorry to be a bother.  When using the use-package command with lsp-mode, do the requirements override quelpa statements in the emacs config?

I uninstalled treemacs completely with the package manager and package-autoremove.

I placed this in my emacs config, prior to lsp stuff:

`
;; treemacs compatiblity fix
(use-package treemacs
     :quelpa t
     (treemacs
       :fetcher github
       :repo "alexander-miller/treemacs"
       :commit "938c2678dc29817f57871445bc25699da0df076d"))
`

But latest treemacs files still keep getting pulled?

**Comment by yyoncho** [comment]
I am not use-package expert, but putting something like this just before the use-package declaration should work:

```
(unless (featurep 'treemacs)
  (quelpa
   :fetcher 'github
   :repo "alexander-miller/treemacs"
   :commit "938c2678dc29817f57871445bc25699da0df076d"))
```

**Comment by yyoncho** [comment]
Also make sure that it runs before any package that requires `treemacs` otherwise it will install the master version again.

**Comment by bridgesense** [comment]
Yeah, I tried compiling that version of treemacs and swapping out the files and lsp-treemacs wouldn't run.  Perhaps I need an older package of that, too?

dap-mode pretty much runs out of the box for a clean emacs install.  This last commit breaks it.  I'm sure there is a way to get this to work, but until there is some documentation to support installation workarounds -- I'd roll this back.

**Comment by bridgesense** [comment]
After tearing out and reinstalling everything, it's just the local variables that no longer show up.  This may be an unrelated issue to another update that happened, because even when I roll back -- it's the local variables that are missing.

**Comment by bridgesense** [comment]
Is this what you were talking about, as far as the Treemacs issue?

![missing local variables](https://i.imgur.com/HGyZfNI.png)

I rolled EVERYTHING back to a couple of months ago.  I compiled all the LSP modules from at least two months ago.  I compiled an earlier version of emacs.

I just updated all the packages in emacs a couple days ago, and didn't have any issues.  Now all the local variables are missing.  It doesn't appear to be a LSP issue, either?

In any case, I thought I could get some context in the issue you were having in treemacs?
